src/neural/dx/network_dx.cc

```diff
@@ -745,18 +782,45 @@ void DxNetwork::Eval(InputsOutputsDx* io, int batch_size) {
 
   // value FC1.
   network_[l++]->Eval(batch_size, tensor_mem_[1], tensor_mem_[0], DXAlloc(),
-                      tensor_mem_[2], tensor_mem_[3], cl);
+                      DXAlloc(), DXAlloc(), cl);
```

Comment:
I made these changes here because it was 'scary' that tensor_mem_[2] was being passed anywhere when it contains the all important residual final output.  Happily fully connecteds don't use scratch at all.

Reply:
Earlier nothing was using values from residual tower after the value head so they were fine to be overwritten, but MLH head would use them, and because FC layers don't use scratch at all the changes are fine.